src/pages/deposit/Deposit.tsx

```diff
@@ -40,13 +40,163 @@ function Deposit({ poolName }: Props): ReactElement | null {
+        if (hasFraxBalance) {
+          setHasFrax(true)
+        }
+      }
+    } catch (err) {
+      console.error('[fetchFraxBalanceOnBSC] ==> err', err)
+    }
+  }
+
+  const fetchFraxBalanceOnMATIC = async () => {
+    try {
+      if (fraxContractOnMATIC && account) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        const hasFraxBalance = (await fraxContractOnMATIC.balanceOf(account)).gt(Zero)
+        if (hasFraxBalance) {
+          setHasFrax(true)
+        }
+      }
+    } catch (err) {
+      console.error('[fetchFraxBalanceOnMATIC] ==> err', err)
+    }
+  }
+
+  const fetchFraxBalanceOnARBITRUM = async () => {
+    try {
+      if (fraxContractOnARBITRUM && account) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        const hasFraxBalance = (await fraxContractOnARBITRUM.balanceOf(account)).gt(Zero)
+        if (hasFraxBalance) {
+          setHasFrax(true)
+        }
+      }
+    } catch (err) {
+      console.error('[fetchFraxBalanceOnARBITRUM] ==> err', err)
+    }
+  }
+
+  const fetchFraxBalanceOnFANTOM = async () => {
+    try {
+      if (fraxContractOnFANTOM && account) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        const hasFraxBalance = (await fraxContractOnFANTOM.balanceOf(account)).gt(Zero)
+        if (hasFraxBalance) {
+          setHasFrax(true)
+        }
+      }
+    } catch (err) {
+      console.error('[fetchFraxBalanceOnFANTOM] ==> err', err)
+    }
+  }
+
+  const fetchFraxBalanceOnMOONRIVER = async () => {
+    try {
+      if (fraxContractOnMOONRIVER && account) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        const hasFraxBalance = (await fraxContractOnMOONRIVER.balanceOf(account)).gt(Zero)
+        if (hasFraxBalance) {
+          setHasFrax(true)
+        }
+      }
+    } catch (err) {
+      console.error('[fraxContractOnMOONRIVER] ==> err', err)
+    }
+  }
+
+  const fetchFraxBalanceOnBOBA = async () => {
+    try {
+      if (fraxContractOnBOBA && account) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        const hasFraxBalance = (await fraxContractOnBOBA.balanceOf(account)).gt(Zero)
+        if (hasFraxBalance) {
+          setHasFrax(true)
+        }
+      }
+    } catch (err) {
+      console.error('[fetchFraxBalanceOnBOBA] ==> err', err)
+    }
+  }
+
+  const fetchFraxBalanceOnHARMONY = async () => {
+    try {
+      if (fraxContractOnHARMONY && account) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        const hasFraxBalance = (await fraxContractOnHARMONY.balanceOf(account)).gt(Zero)
+        if (hasFraxBalance) {
+          setHasFrax(true)
+        }
+      }
+    } catch (err) {
+      console.error('[fetchFraxBalanceOnHARMONY] ==> err', err)
+    }
+  }
+
+  
+  useEffect(() => {
+    const fetchFraxBalance = () => {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
```

Comment:
This is not what we want to do here, please find a better way to handle this call in a more logical manner.